core/services/relay/evm/chain_reader.go

```diff
@@ -178,8 +220,51 @@ func (cr *chainReader) GetMaxDecodingSize(ctx context.Context, n int, itemType s
 	return 0, fmt.Errorf("Unimplemented method GetMaxDecodingSize called %w", relaytypes.ErrorChainReaderUnsupported{})
 }
 
+// GetLatestValue retrieves latest value from contract.
 func (cr *chainReader) GetLatestValue(ctx context.Context, bc relaytypes.BoundContract, method string, params any, returnVal any) error {
-	return fmt.Errorf("Unimplemented method GetLatestValue called %w", relaytypes.ErrorChainReaderUnsupported{})
+	var response []byte
+	chainContractReader, exists := cr.chainContractReaders[bc.Name]
+	if !exists {
+		return fmt.Errorf("chain reading not defined for:%s on contract:%s", method, bc.Name)
+	}
+	chainReadingDefinition := chainContractReader.ChainReaderDefinitions[method]
+	chainSpecificName := chainReadingDefinition.ChainSpecificName
+	contractAddr := common.HexToAddress(bc.Address)
+
+	if chainReadingDefinition.ReadType == types.Method {
+		callData, err := chainContractReader.ParsedContractABI.Pack(chainSpecificName, params)
+		if err != nil {
+			return err
+		}
+
+		// if params are nil, repack callData because nil != empty
+		if params == nil {
+			callData, err = chainContractReader.ParsedContractABI.Pack(chainSpecificName)
+			if err != nil {
+				return err
+			}
+		}
+
+		ethCallMsg := ethereum.CallMsg{
+			From: common.Address{},
+			To:   &contractAddr,
+			Data: callData,
+		}
+
+		response, err = cr.chain.Client().CallContract(ctx, ethCallMsg, nil)
```

Comment:
Have we researched this enough to know that by making direct calls to `CallContract` we're not missing out on any features we might have been getting from using gethwrappers? If so this is great, just a bit nervous of changing how it works if there might be other things gethwrapper in case there are any other things gethwrappers is doing for us.

Reply:
I don't see any issues with doing it like this especially for the functionalities that we need.